### PR TITLE
Add EmitC support for conversion ops

### PR DESCRIPTION
**Pull request by dajuro**
Further adds tests for `vm.trunc` ops to the VM.

Co-authored-by: Marius Brehler <marius.brehler@iml.fraunhofer.de>

### CODE REVIEW

**Review by marbre — APPROVED** [review]

**Review by benvanik — CHANGES_REQUESTED** [review]
Nice - looks good! Can you please also swap out the `iree/vm/bytecode_dispatch.c` implementations with calls to these new functions?

This macro: https://github.com/google/iree/blob/e592193bcb003dd10bd458ce9ac89db4bc4fd9a6/iree/vm/bytecode_dispatch.c#L1020-L1032

Should be changed to take the op_func name and reference your new ops.h functions as with the above DISPATCH_OP_CORE_UNARY_ALU_I32 macro.

File: iree/vm/bytecode_dispatch.c [hunk 1]
```diff
@@ -1007,19 +1007,19 @@ iree_status_t iree_vm_bytecode_dispatch(
...
-#define DISPATCH_OP_CORE_CAST_I32(op_name, src_type, dst_type) \
-  DISPATCH_OP(CORE, op_name, {                                 \
-    int32_t operand = VM_DecOperandRegI32("operand");          \
-    int32_t* result = VM_DecResultRegI32("result");            \
-    *result = (dst_type)((src_type)operand);                   \
+#define DISPATCH_OP_CORE_CAST_I32(op_name, op_func)   \
+  DISPATCH_OP(CORE, op_name, {                        \
+    int32_t operand = VM_DecOperandRegI32("operand"); \
+    int32_t* result = VM_DecResultRegI32("result");   \
+    *result = op_func(operand);                       \
```
Comment (marbre) [inline iree/vm/bytecode_dispatch.c:1014]: No need for a separate `DISPATCH_OP_CORE_CAST_I32` macro any longer. As Ben suggested, you can use the already existing `DISPATCH_OP_CORE_UNARY_ALU_I32` macro.

**Review by marbre — CHANGES_REQUESTED** [review]

File: iree/vm/bytecode_dispatch.c (hunk 1 above)
Comment (marbre) [reply iree/vm/bytecode_dispatch.c:1014]: After our discussion, we should clean this up in a follow up and define "common" macros as soon as we refactored the dispatcher to use shared implementations.

**Review by marbre — APPROVED** [review]

File: iree/vm/bytecode_dispatch.c (hunk 1 above)
Comment (dajuro) [reply iree/vm/bytecode_dispatch.c]: I removed `DISPATCH_OP_CORE_CAST_I32` in favor of `DISPATCH_OP_CORE_UNARY_ALU_I32`. I suggest to reduce the number of macros by unifying the macros, e.g. `DISPATCH_OP_CORE_UNARY_I32` or `DISPATCH_OP_CORE_BINARY_I32`. The less macros - the better :)

Comment (marbre) [reply iree/vm/bytecode_dispatch.c]: Thanks for adding a TODO @dajuro. Let's handle cleaning up the macros in a follow up :)

**Review by benvanik — APPROVED** [review]

**Pull request merged**